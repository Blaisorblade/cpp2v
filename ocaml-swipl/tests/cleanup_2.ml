open SWI_prolog.API

let _ =
  initialise [|Sys.argv.(0); "-q"|];
  for i = 0 to 3 do
    Printf.printf "Init... (%i)\n%!" i;
    initialise [|Sys.argv.(0); "-q"|];
    Printf.printf "Cleanup...\n%!";
    try
      cleanup ~no_reclaim_memory:true ~no_cancel:false ~status:0;
      Printf.printf "OK...\n%!"
    with
    | Cleanup(`Cancelled) -> Printf.printf "Cancelled...\n%!"
    | Cleanup(`Failed   ) -> Printf.printf "Failed...\n%!"
    | Cleanup(_         ) -> Printf.printf "Recursive...\n%!"
  done
