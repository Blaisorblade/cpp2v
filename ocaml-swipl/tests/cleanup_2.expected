Init... (0)
Cleanup...
OK...
Init... (1)
Cleanup...
OK...
Init... (2)
Cleanup...
OK...
Init... (3)
Cleanup...
OK...
